docs/_includes/docker.md

```diff
@@ -16,8 +16,66 @@ Since Summon has pluggable providers, you aren't locked into any one solution fo
 managing your secrets.
 
 Summon makes it easy to inject secrets as environment variables into your Docker
-containers by taking advantage of Docker's `--env-file` argument. This is done
-on-demand by using the variable `@SUMMONENVFILE` in the arguments of the process
+containers by taking advantage of Docker's CLI arguments (`--env-file` or, `--env` and `--volume`. There are two options available. It's possible to mix and match as you see fit.
+
+## --env and --volume arguments
+This is done on-demand by using the variable `@SUMMONDOCKERARGS` in the arguments of the process
+you are running with Summon. This variable is replaced by combinations of the Docker arguments `--env` and `--volume` such that the secrets injected by summon are passed into the Docker container. The `--volume` arguments allow memory-mapped temporary files from variables with the `!file` tag to be resolvable inside the container.
+
+**NOTE:** Using the `!file` tag with `@SUMMONDOCKERARGS` assumes that the Docker CLI is being run on the host that is used to create volume mounts to the container. For when this is not the case simply avoid using the `!file` tag, but be mindful that in that case you lose the benefits of memory-mapped temporary files.
+
+```bash
+$ summon -p keyring.py -D env=dev docker run @SUMMONDOCKERARGS deployer
+Checking credentials
+Deploying application
+```
+
+### Example
+The example below demonstrates the use @SUMMONDOCKERARGS. For the sake of brevity
+we use an inline `secrets.yml` and the `/bin/echo` provider. Some points to note:
+1. `summon` is
+invoking docker as the child process.
+2. `@SUMMONDOCKERARGS` is replaced with a combination of `--env` and `--volume`
+arguments.
+3. Variable `D` uses the `!file` tag and therefore is the only one that
+results in a `--volume` argument. The path to this variable inside the container
+is as it is on the host.
```

Comment:
Good point. This might get tricky if for host to container you're going from Windows to Linux or vice-versa. I'm wondering now if I should just altogether disregard the case for files since it adds complication. Having `--env` would already be a massive improvement.

Reply:
I like the idea of allowing the user to specify the location that Summon creates files. Even though not every platform supports `/dev/shm`, the benefit of summon creating files is that it is also responsible for cleaning them up... as much as possible we should retain this for the users in the new features we add. 

For now I got rid of the implicit `--volume` arguments. We'll let the user do that themselves. Some things to note
1. Summon does have a way to configure the tmp directory but it's difficult for the user to predict which one is being used. See https://github.com/cyberark/summon/blob/master/internal/command/temp_factory.go#L29. We try `/dev/shm` then os home directory then os tmp directory. It's not great!
2. I suppose we should just give the user a way to more explicitly specify this directory, then they can volume mount themselves. However there'd still be an issue of going from say Windows to Linux Containers, and vice versa. I suppose we'll need to look into this more.